extra/modules/confiant-ad-quality/src/main/java/org/prebid/server/hooks/modules/com/confiant/adquality/v1/ConfiantAdQualityBidResponsesScanHook.java

```diff
@@ -0,0 +1,121 @@
+import org.prebid.server.auction.PrivacyEnforcementService;
+import org.prebid.server.auction.model.AuctionContext;
+import org.prebid.server.auction.model.BidderResponse;
+import org.prebid.server.hooks.execution.v1.bidder.AllProcessedBidResponsesPayloadImpl;
+import org.prebid.server.hooks.modules.com.confiant.adquality.core.BidsMapper;
+import org.prebid.server.hooks.modules.com.confiant.adquality.core.BidsScanResult;
+import org.prebid.server.hooks.modules.com.confiant.adquality.core.RedisClient;
+import org.prebid.server.hooks.modules.com.confiant.adquality.core.RedisScanStateChecker;
+import org.prebid.server.hooks.modules.com.confiant.adquality.model.OperationResult;
+import org.prebid.server.hooks.modules.com.confiant.adquality.v1.model.InvocationResultImpl;
+import org.prebid.server.hooks.v1.InvocationAction;
+import org.prebid.server.hooks.v1.InvocationResult;
+import org.prebid.server.hooks.v1.InvocationStatus;
+import org.prebid.server.hooks.v1.auction.AuctionInvocationContext;
+import org.prebid.server.hooks.v1.bidder.AllProcessedBidResponsesHook;
+import org.prebid.server.hooks.v1.bidder.AllProcessedBidResponsesPayload;
+
+import java.util.List;
+
+public class ConfiantAdQualityBidResponsesScanHook implements AllProcessedBidResponsesHook {
+
+    private static final String CODE = "confiant-ad-quality-bid-responses-scan-hook";
+
+    private final RedisClient redisClient;
+
+    private final RedisScanStateChecker redisScanStateChecker;
+
+    private final PrivacyEnforcementService privacyEnforcementService;
+
+    public ConfiantAdQualityBidResponsesScanHook(
+            RedisClient redisClient,
+            RedisScanStateChecker redisScanStateChecker,
+            PrivacyEnforcementService privacyEnforcementService) {
+        this.redisClient = redisClient;
+        this.redisScanStateChecker = redisScanStateChecker;
+        this.privacyEnforcementService = privacyEnforcementService;
+    }
+
+    @Override
+    public Future<InvocationResult<AllProcessedBidResponsesPayload>> call(
+            AllProcessedBidResponsesPayload allProcessedBidResponsesPayload,
+            AuctionInvocationContext auctionInvocationContext) {
+        final BidRequest bidRequest = getBidRequest(auctionInvocationContext);
+        final List<BidderResponse> responses = allProcessedBidResponsesPayload.bidResponses();
+        final boolean isScanDisabled = redisScanStateChecker.isScanDisabled();
+
+        if (isScanDisabled) {
+            return getResult(new BidsScanResult(OperationResult.empty()), auctionInvocationContext);
+        }
+
+        final Promise<InvocationResult<AllProcessedBidResponsesPayload>> processed = Promise.promise();
+
+        redisClient.submitBids(BidsMapper.bidResponsesToRedisBids(bidRequest, responses))
+                .onComplete(result -> processed
+                        .complete(getResult(result.result(), auctionInvocationContext).result()));
+
+        return processed.future();
```

Comment:
After removing `Future` from previous comment, we can simplify this block of code:
```java 
 return redisClient.submitBids(BidsMapper.bidResponsesToRedisBids(bidRequest, responses))
                .map(scanResult -> getResult(scanResult, auctionInvocationContext)));
  ```